app_dart/lib/src/request_handlers/update_task_status.dart

```diff
@@ -178,8 +178,15 @@ class UpdateTaskStatus extends ApiRequestHandler<UpdateTaskStatusResponse> {
   }
 
   /// Construct the Datastore key for [Commit] that is the ancestor to this [Task].
-  Key _constructCommitKey(DatastoreService datastore) {
-    final String id = 'flutter/flutter/${requestData[gitBranchParam]}/${requestData[gitShaParam]}';
+  ///
+  /// Throws [BadRequestException] if the given git branch does not exist in [CocoonConfig].
+  Future<Key> _constructCommitKey(DatastoreService datastore) async {
+    final String gitBranch = requestData[gitBranchParam] as String;
+    final List<String> flutterBranches = await config.flutterBranches;
+    if (!flutterBranches.contains(gitBranch)) {
+      throw BadRequestException('$gitBranch is not a flutter/flutter branch');
```

Comment:
how does this exception get surfaced? Will this register as a failed task?

Reply:
Yes, red task. The test runner will output this error message as the reason why the task failed. After the migration, this should only be possibly encountered on new release branches.

Reply:
Cool. Can you update the exception message to explicitly state that `'$gitBranch is not defined in https://github.com/flutter/cocoon/blob/master/app_dart/dev/branches.txt'`?

Reply:
Good point! Added, but there's a bit more steps (like flush the cache) so I linked to https://github.com/flutter/cocoon/tree/master/app_dart#branching-support-for-flutter-repo

I thought about pinning to a commit sha, but it got really long. Do you think a flutter.dev/go/ link would be appropriate?

Reply:
I think linking to the master tree is probably better than pinning it. And if we change the anchor name, the link still sends you to the readme.